src/core/model/table/typed_column_data.h

```diff
@@ -40,6 +40,15 @@ class TypedColumnData : public model::AbstractColumnData {
 
     friend class TypedColumnDataFactory;
 
+    void Destruct(TypeId type_id, std::byte const* value) {
+        if (type_id == +TypeId::kString || type_id == +TypeId::kBigInt) {
+            Type::Destruct<StringType>(value);
+        }
+        if (type_id == +TypeId::kDate) {
+            Type::Destruct<DateType>(value);
+        }
+    }
```

Comment:
It's the same as in `MixedType`, why implementing it twice?
I think it should be made `static` and moved to `Type` instead of unneeded templated version of `Destruct` and used everywhere

Reply:
Seems like adding such method to 'Type' won't work as we can't use functionality of a child class in the base class. I'll try to move destruct to a separate header.

Reply:
Separate header seems OK for now